Provide How-to Guide for bootstrapping and connecting to the new device (#102)
eclipse-kanto/kanto: web/site/content/docs/how-to-guides/bootstrap-device.md

---
title: "Bootstrap device"
type: docs
description: >
    Automatically provision your device via bootstrapping.
weight: 5
---

> Following the steps below you will automatically provision a new device via a publicly available Eclipse Hono sandbox using Eclipse Kanto.

Reviewer: ```suggestion
By following the steps below you will automatically provision a new device via a publicly available Eclipse Hono sandbox using Eclipse Kanto.
```

A simple Eclipse Hono northbound business application written in Python is

> provided to explore the capabilities for device bootstrapping and automatically provision new device.

Reviewer: ```suggestion
provided to explore the capabilities for device bootstrapping and automatically provision а new device.
```

The custom Python application handles the request by automatically
provisioning a new device. Upon successful automatically provisioning it sends back
all mandatory remote communication, identification and authentication data.
On the edge side, the response is handled by updating the connection configuration with the received data
and by executing a basic
{{% refn "https://github.com/eclipse-kanto/kanto/blob/main/quickstart/post_bootstrap.sh" %}}`post_bootstrap.sh`{{% /refn %}}
script to restart the Suite Connector service for the changes to take effect.

### Before you begin

To ensure that your edge device is capable to execute the steps in this guide, you need:

* If you don't have an installed and running Eclipse Kanto, follow {{% relrefn "install" %}} Install Eclipse Kanto {{% /relrefn %}}
* If you don't have a connected Eclipse Kanto to Eclipse Hono sandbox,
  follow {{% relrefn "hono" %}} Explore via Eclipse Hono {{% /relrefn %}}
* The {{% refn "https://github.com/eclipse-kanto/kanto/blob/main/quickstart/hono_commands_sb.py" %}}
  suite bootstrapping application {{% /refn %}} and {{% refn "https://github.com/eclipse-kanto/kanto/blob/main/quickstart/post_bootstrap.sh" %}} 

>   post bootstrap {{% /refn %}}

Reviewer: ```suggestion
  post bootstrap script{{% /refn %}}
```

  Navigate to the `quickstart` folder where the resources from the {{% relrefn "hono" %}} Explore via Eclipse Hono {{% /relrefn %}}
  guide are located and execute the following script:

  ```shell
  wget https://github.com/eclipse-kanto/kanto/raw/main/quickstart/hono_commands_sb.py && \
  wget https://github.com/eclipse-kanto/kanto/raw/main/quickstart/post_bootstrap.sh
  ```
* Grab the post script file and place it in the /var/tmp/suite-bootstrapping directory via executing:

  ```shell
  sudo mkdir -p /var/tmp/suite-bootstrapping/ && sudo cp ./post_bootstrap.sh /var/tmp/suite-bootstrapping/
  ```
* Back up `/ect/suite-connector/config.json` as this file will be modified from this guide
* Stop suite-connector.service. Suite Connector and Suite Bootstrapping cannot work together
  ```shell
  sudo systemctl stop suite-connector.service
  ```

### Configure Suite Bootstrapping

> Open file `/etc/suite-connector/config.json`, copy `tenantId`, `deviceId`, `authId` and `password`

Reviewer: ```suggestion
Open file `/etc/suite-connector/config.json`, copy `tenantId`, `deviceId`, `authId` and `password`.
```

```json
{
    ...
    "tenantId": "demo",
    "deviceId": "demo:device",
    "authId": "demo_device",
    "password": "secret"
    ...
}
```
Eclipse Kanto uses the `/etc/suite-bootstrapping/config.json` to acquire all the remote communication, identification and
authentication data to establish the remote connection for bootstrapping.
It is also where you need to specify the path to the post bootstrapping script and where to store received response data.

> Update it with previous copied settings and with following:

Reviewer: ```suggestion
Update the configuration as shown below and replace `tenantId`, `deviceId`, `authId` and `password` with the settings that you copied in the previous step.
```
